openfisca_core/formulas.py

```diff
@@ -834,110 +823,71 @@ def new_filled_column(
-            if dated_formula_class['stop_instant'] is None:
-                dated_formula_class['stop_instant'] = next_dated_formula_class['start_instant'].offset(-1, 'day')
-            else:
-                assert dated_formula_class['stop_instant'] < next_dated_formula_class['start_instant'], \
-                    "Dated formulas overlap: {} & {}".format(dated_formula_class, next_dated_formula_class)
-
-        # Add dated formulas defined in (optional) reference column when they are not overridden by new dated
-        # formulas.
-        if reference_column is not None and issubclass(reference_column.formula_class, DatedFormula):
-            for reference_dated_formula_class in reference_column.formula_class.dated_formulas_class:
-                reference_dated_formula_class = reference_dated_formula_class.copy()
-                for dated_formula_class in dated_formulas_class:
-                    if reference_dated_formula_class['start_instant'] == dated_formula_class['start_instant'] \
-                            and reference_dated_formula_class['stop_instant'] == dated_formula_class[
-                                'stop_instant']:
-                        break
-                    if reference_dated_formula_class['start_instant'] >= dated_formula_class['start_instant'] \
-                            and reference_dated_formula_class['start_instant'] < dated_formula_class[
-                                'stop_instant']:
-                        reference_dated_formula_class['start_instant'] = dated_formula_class['stop_instant'].offset(
-                            1, 'day')
-                    if reference_dated_formula_class['stop_instant'] > dated_formula_class['start_instant'] \
-                            and reference_dated_formula_class['stop_instant'] <= dated_formula_class[
-                                'stop_instant']:
-                        reference_dated_formula_class['stop_instant'] = dated_formula_class['start_instant'].offset(
-                            -1, 'day')
-                    if reference_dated_formula_class['start_instant'] > reference_dated_formula_class[
-                            'stop_instant']:
-                        break
-                else:
-                    dated_formulas_class.append(reference_dated_formula_class)
-            dated_formulas_class.sort(key = lambda dated_formula_class: dated_formula_class['start_instant'])
+    # Turn function into a dated function
+    formula_name_prefix = 'formula'
+    formula_name_separator = '_'
+    dated_formulas_class = []
+    for function_name, function in specific_attributes.copy().iteritems():
+        if not function_name.startswith(formula_name_prefix):
+            # Current attribute isn't a formula
+            continue
+
+        formula_start_date = extract_formula_date_from_name(function_name, formula_name_prefix, formula_name_separator)
+        if end is not None:
+            end_date = get_datetime_date(name, end)
```

Comment:
You're parsing the string `n` times. See above the suggestion to parse it once and for all.